tests/components/alert/test_init.py

```diff
@@ -117,239 +90,249 @@ def async_toggle(hass, entity_id):
-        events = self._setup_notify()
-
-        config = deepcopy(TEST_CONFIG)
-        config[alert.DOMAIN][NAME][alert.CONF_DONE_MESSAGE] = TEMPLATE
-        assert setup_component(self.hass, alert.DOMAIN, config)
-
-        self.hass.states.set(TEST_ENTITY, STATE_ON)
-        self.hass.block_till_done()
-        self.hass.states.set(TEST_ENTITY, STATE_OFF)
-        self.hass.block_till_done()
-        self.assertEqual(2, len(events))
-        last_event = events[-1]
-        self.assertEqual(last_event.data[notify.ATTR_MESSAGE], TEST_ENTITY)
-
-    def test_sending_titled_notification(self):
-        """Test notifications."""
-        events = self._setup_notify()
-
-        config = deepcopy(TEST_CONFIG)
-        config[alert.DOMAIN][NAME][alert.CONF_TITLE] = TITLE
-        assert setup_component(self.hass, alert.DOMAIN, config)
-
-        self.hass.states.set(TEST_ENTITY, STATE_ON)
-        self.hass.block_till_done()
-        self.assertEqual(1, len(events))
-        last_event = events[-1]
-        self.assertEqual(last_event.data[notify.ATTR_TITLE], TEST_TITLE)
-
-    def test_sending_data_notification(self):
-        """Test notifications."""
-        events = self._setup_notify()
-
-        config = deepcopy(TEST_CONFIG)
-        config[alert.DOMAIN][NAME][alert.CONF_DATA] = TEST_DATA
-        assert setup_component(self.hass, alert.DOMAIN, config)
-
-        self.hass.states.set(TEST_ENTITY, STATE_ON)
-        self.hass.block_till_done()
-        self.assertEqual(1, len(events))
-        last_event = events[-1]
-        self.assertEqual(last_event.data[notify.ATTR_DATA], TEST_DATA)
-
-    def test_skipfirst(self):
-        """Test skipping first notification."""
-        config = deepcopy(TEST_CONFIG)
-        config[alert.DOMAIN][NAME][alert.CONF_SKIP_FIRST] = True
-        events = []
-
-        @callback
-        def record_event(event):
-            """Add recorded event to set."""
-            events.append(event)
-
-        self.hass.services.register(notify.DOMAIN, NOTIFIER, record_event)
-
-        assert setup_component(self.hass, alert.DOMAIN, config)
-        assert len(events) == 0
-
-        self.hass.states.set("sensor.test", STATE_ON)
-        self.hass.block_till_done()
-        assert len(events) == 0
-
-    def test_noack(self):
-        """Test no ack feature."""
-        entity = alert.Alert(self.hass, *TEST_NOACK)
-        self.hass.add_job(entity.begin_alerting)
-        self.hass.block_till_done()
-
-    def test_done_message_state_tracker_reset_on_cancel(self):
-        """Test that the done message is reset when canceled."""
-        entity = alert.Alert(self.hass, *TEST_NOACK)
-        entity._cancel = lambda *args: None
-        assert entity._send_done_message is False
-        entity._send_done_message = True
-        self.hass.add_job(entity.end_alerting)
-        self.hass.block_till_done()
-        assert entity._send_done_message is False
+def _setup_notify(hass):
```

Comment:
We should use a fixture for this.

```python
@pytest.fixture
def mock_notifier(hass):
    events = []

    @callback
    def record_event(event):
        """Add recorded event to set."""
        events.append(event)

    hass.services.async_register(notify.DOMAIN, NOTIFIER, record_event)

    return events
```

Now in your tests, you can add `mock_notifier` to the args of the test to set it up automatically.

```python
# old
async def test_fire(hass):

# new
async def test_fire(hass, mock_notifier):
```

The value of `mock_notifier` will be what was returned from the fixture (`events`)